src/components/transactions/Emode/EmodeModalContent.tsx

```diff
@@ -49,50 +49,59 @@ export const EmodeModalContent = () => {
 
   // get all emodes
   useEffect(() => {
-    const emodeCategoriesArray: EmodeCategory[] = [];
-    reserves.forEach((reserve) => {
-      const emodeFound = emodeCategoriesArray.find(
-        (category) => category.id === reserve.eModeCategoryId
-      );
-      if (!emodeFound) {
-        const emodeParams: EmodeCategory = {
-          id: reserve.eModeCategoryId,
-          ltv: reserve.eModeLtv,
-          liquidationThreshold: reserve.eModeLiquidationThreshold,
-          liquidationBonus: reserve.eModeLiquidationBonus,
-          priceSource: reserve.eModePriceSource,
-          label: reserve.eModeLabel,
-          assets: [],
-        };
-
-        // get all emode asets
-        reserves.forEach((eReserve) => {
-          if (eReserve.eModeCategoryId === reserve.eModeCategoryId) {
-            emodeParams.assets.push(eReserve.symbol);
-          }
-        });
-
-        emodeCategoriesArray.push(emodeParams);
+    if (!selectedEmode) {
+      const emodeCategoriesArray: EmodeCategory[] = [];
+      reserves.forEach((reserve) => {
+        const emodeFound = emodeCategoriesArray.find(
+          (category) => category.id === reserve.eModeCategoryId
+        );
+        if (!emodeFound) {
+          const emodeParams: EmodeCategory = {
+            id: reserve.eModeCategoryId,
+            ltv: reserve.eModeLtv,
+            liquidationThreshold: reserve.eModeLiquidationThreshold,
+            liquidationBonus: reserve.eModeLiquidationBonus,
+            priceSource: reserve.eModePriceSource,
+            label: reserve.eModeLabel,
+            assets: [],
+          };
+
+          // get all emode asets
+          reserves.forEach((eReserve) => {
+            if (eReserve.eModeCategoryId === reserve.eModeCategoryId) {
+              emodeParams.assets.push(eReserve.symbol);
+            }
+          });
+
+          emodeCategoriesArray.push(emodeParams);
+        }
+      });
+      const emodeCategories: Record<number, EmodeCategory> = {};
+      emodeCategoriesArray.forEach((category) => {
+        emodeCategories[category.id] = category;
+      });
+      // Default settings of the modal window, depending on the user's active emode and # of available categories
+      const selectedEmodeId =
```

Comment:
One (non-blocking) improvement I can see that would be helpful is to have an enum for use of the `user.userEmodeCategoryId`, so that statements like these and throughout the React component JSX more readable.

Reply:
Would have side-effects and improve other areas like [here](https://github.com/aave/interface/blob/553342f5def9fc5fec087bdd829a85c42dca55ac/src/components/transactions/Emode/EmodeNaming.ts#L1)

Reply:
Currently, we export a helper function: https://github.com/aave/interface/blob/main/src/components/transactions/Emode/EmodeNaming.ts to get the name based on id since the same id might be named differently on different networks